### PR TITLE
Changing test_create_pool_block_pool.py to 4.9 and check compression efficiency in UI

### PR DESCRIPTION
**Pull request by shyRozen**
Update the testcase for 4.9 with different location for block_pool page, handling default binding mode for storageclass and checking compression efficiency from UI with prometheus values.
Signed-off-by: Shay Rozen <shay.rozen@gmail.com>

### CODE REVIEW

**Comment by shyRozen** [comment]
Added the need testing although job passed it is for 4.9.
As the test is also for 4.8 to test it. There is already a job for it but it is waiting for psi
[validation for 4.8](https://ocs4-jenkins-csb-ocsqe.apps.ocp4.prod.psi.redhat.com/job/qe-trigger-test-pr/1862/)

**Comment by shyRozen** [comment]
First validation is for 4.9. Second for 4.8 aborted because PSI retry. 
Third validation for 4.8 aws.

File: ocs_ci/ocs/cluster.py [hunk 1]
```diff
@@ -1182,6 +1182,25 @@ def validate_compression(pool_name):
...
+    Returns:
+        pool id if found else None
+
+    """
+    ceph_df_detail_output = get_ceph_df_detail()
+    pool_list = ceph_df_detail_output.get("pools")
+    for pool in pool_list:
+        if pool.get("name") == pool_name:
+            return pool.get("id")
+    return None
```
Comment (ebenahar) [inline ocs_ci/ocs/cluster.py:1201]: I think it isn't needed

File: ocs_ci/ocs/ui/block_pool.py [hunk 2]
```diff
@@ -173,3 +176,130 @@ def reach_pool_limit(self, replica, compression):
...
+    ):
+        """
+        Get pool compression efficiency from UI and compare them against prometheus metrics
+
+        Args:
+            pool_name (str): The pool name to be checked.
+            expected_compression_saving (str): The expected compression saving in string like "1.5 GiB"
+
+        Returns:
+            (bool): True if UI and prometheus parameters are equal to expected_compression_saving - else False.
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/block_pool.py:191]: replace `(bool)` with `bool`

File: ocs_ci/ocs/ui/block_pool.py [hunk 3]
```diff
@@ -173,3 +176,130 @@ def reach_pool_limit(self, replica, compression):
...
+            f" compression_saving {compression_saving}"
+        )
+
+        # Get pool id and query Prometheus
+        pool_id = get_pool_id(pool_name)
+        if pool_id is None:
+            raise PoolIdNotFound(f"Pool {pool_name} didn't return pool id")
+        new_prom = PrometheusAPI()
+        ceph_pool_compress_bytes_used_data = new_prom.query(
+            f'ceph_pool_compress_bytes_used{{pool_id="{pool_id}"}}'
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/block_pool.py:228]: please save the query in ocs_ci/ocs/constants.py:1227

File: ocs_ci/ocs/ui/block_pool.py [hunk 4]
```diff
@@ -173,3 +176,130 @@ def reach_pool_limit(self, replica, compression):
...
+        )
+        ceph_pool_compress_under_bytes_data = new_prom.query(
+            f'ceph_pool_compress_under_bytes{{pool_id="{pool_id}"}}'
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/block_pool.py:231]: same

File: ocs_ci/ocs/ui/block_pool.py [hunk 5]
```diff
@@ -173,3 +176,130 @@ def reach_pool_limit(self, replica, compression):
...
+            return_value_are_equal = False
+            logger.info(
+                f"UI compression saving are not equal to expected saving: "
+                f"UI compression value is {compression_saving_value} "
+                f"and expected compression value is {expected_compression_value}. "
+                f"UI compression unit is {compression_saving_unit} "
+                f"and prometheus compression unit is {expected_compression_unit}"
+            )
+
+        if (
```
Comment (ebenahar) [inline ocs_ci/ocs/ui/block_pool.py:278]: shouldn't it be `elif`? do we want to have all these conditions checked? if the first sets the `return_value_are_equal` to False, isn't it enough?

File: ocs_ci/utility/utils.py [hunk 6]
```diff
@@ -2653,6 +2654,26 @@ def convert_device_size(unformatted_size, units_to_covert_to):
     return conversion[units_to_covert_to][units]
 
 
+def convert_bytes_into_right_measure(size_bytes):
```
Comment (ebenahar) [inline ocs_ci/utility/utils.py:2657]: isn't it similar to ocs_ci.utility.utils.convert_device_size?

File: ocs_ci/ocs/cluster.py (hunk 1 above)
Comment (shyRozen) [reply ocs_ci/ocs/cluster.py:1201]: Correct

File: ocs_ci/ocs/ui/block_pool.py (hunk 5 above)
Comment (shyRozen) [reply ocs_ci/ocs/ui/block_pool.py:278]: I thought to get a wider picture on all value compare before returning False. It doesn't take time as data is not retrieved but just compared.

File: ocs_ci/utility/utils.py (hunk 6 above)
Comment (shyRozen) [reply ocs_ci/utility/utils.py:2657]: In convert_device_size you have to decide the right measure unit like GI MI TI etc. In my function the measure unit is self calculate for the size. 
My gets only bytes which is what ceph gives in json and convert_device_size is getting any kind of measure. 
convert_bytes_into_right_measure is predicting the measure unit by the size of the input. Similar but different. 
If you don't know what units the outcome will be (GB TB PB etc) it's better to use mine.

File: ocs_ci/ocs/ui/block_pool.py (hunk 3 above)
Comment (shyRozen) [reply ocs_ci/ocs/ui/block_pool.py:228]: I saw the queries in constant. None of them have variables inside them and mine have pool id.
Whomever will use them will have to know that he needs to format the string to insert the variable. Are you ok with that?

File: ocs_ci/utility/utils.py (hunk 6 above)
Comment (ebenahar) [reply ocs_ci/utility/utils.py:2657]: ok

File: ocs_ci/ocs/ui/block_pool.py (hunk 3 above)
Comment (ebenahar) [reply ocs_ci/ocs/ui/block_pool.py:228]: yes, it is ok

**Pull request closed without merging**